src/sinks/util/http.rs

```diff
@@ -149,8 +149,18 @@ where
         let mut http = HttpConnector::new_with_resolver(resolver.clone());
         http.enforce_http(false);
 
-        let tls = tls_connector_builder(&tls_settings.into())?;
-        let https = HttpsConnector::with_connector(http, tls)?;
+        let settings = tls_settings.into();
+        let tls = tls_connector_builder(&settings)?;
+        let mut https = HttpsConnector::with_connector(http, tls)?;
+
+        let settings = settings.tls().map(Clone::clone);
```

Comment:
Isn't this just `let settings = settings.tls().clone();`?

Reply:
Ok, so this one is weird and I am not 100% sure why, but for some reason the compiler is having trouble figuring out the borrow when you use clone. The map solution works because it goes from `Option<&T>` -> `Option<T>` but clone doesn't do that because I believe it wants to clone the &T so you get another &T instead of doing an inner clone.

`Option::cloned` works though!